File: app/src/main/java/com/example/bloodbank/data/local/SharedPreferencesManger.java
package com.example.bloodbank.data.local;

import android.app.Activity;
import android.content.Context;
import android.content.SharedPreferences;

import com.example.bloodbank.data.model.login.ClientData;
import com.google.gson.Gson;

public class SharedPreferencesManger {

    private static SharedPreferences sharedPreferences = null;
    public static String USER_DATA = "USER_DATA";
    public static String PHONE = "PHONE";
    public static String API_TOKEN = "API_TOKEN";
    public static String PASSWORD = "PASSWORD";


    public static void setSharedPreferences(Activity activity) {
        if (sharedPreferences == null) {
            sharedPreferences = activity.getSharedPreferences(
                    "Blood", Context.MODE_PRIVATE);
        }
    }

    public static void SaveData(Activity activity, String data_Key, String data_Value) {
        if (sharedPreferences != null) {
            SharedPreferences.Editor editor = sharedPreferences.edit();
            editor.putString(data_Key, data_Value);
            editor.commit();
        } else {
            setSharedPreferences(activity);
        }
    }

    public static void SaveData(Activity activity, String data_Key, boolean data_Value) {
        if (sharedPreferences != null) {
            SharedPreferences.Editor editor = sharedPreferences.edit();
            editor.putBoolean(data_Key, data_Value);
            editor.commit();
        } else {
            setSharedPreferences(activity);
        }
    }

    public static String LoadData(Activity activity, String data_Key) {
        if (sharedPreferences != null) {
            SharedPreferences.Editor editor = sharedPreferences.edit();
        } else {
            setSharedPreferences(activity);
        }

        return sharedPreferences.getString(data_Key, null);
    }

    public static boolean LoadBoolean(Activity activity, String data_Key) {
        if (sharedPreferences != null) {
            SharedPreferences.Editor editor = sharedPreferences.edit();
        } else {
            setSharedPreferences(activity);
        }

        return sharedPreferences.getBoolean(data_Key, false);
    }

    public static void SaveData(Activity activity, String data_Key, Object data_Value) {
        setSharedPreferences(activity);
        if (sharedPreferences != null) {
            SharedPreferences.Editor editor = sharedPreferences.edit();
            Gson gson = new Gson();
            String StringData = gson.toJson(data_Value);
            editor.putString(data_Key, StringData);
            editor.commit();
        }
    }


    public static ClientData loadUserData(Activity activity) {
        setSharedPreferences(activity);

        ClientData loginData = null;
        Gson gson = new Gson();
        loginData = gson.fromJson(LoadData(activity, USER_DATA), ClientData.class);

        return loginData;
    }

    public static void clean(Activity activity) {
        setSharedPreferences(activity);
        if (sharedPreferences != null) {
            SharedPreferences.Editor editor = sharedPreferences.edit();
            editor.clear();
            editor.commit();
        }
    }

}

File: app/src/main/java/com/example/bloodbank/data/local/SharedPreferencesManger.kt
package com.example.bloodbank.data.local

import android.content.Context
import android.content.Context.MODE_PRIVATE
import android.content.SharedPreferences
import com.example.bloodbank.data.model.login.ClientData
import com.example.bloodbank.utils.API_TOKEN
import com.example.bloodbank.utils.USER_DATA
import com.google.gson.Gson

open class SharedPreferencesManger(var context: Context) {
    private var sharedPreferences: SharedPreferences? = null

    init {
        if (sharedPreferences == null) {
            sharedPreferences = context.getSharedPreferences(
                    "MyBlood", MODE_PRIVATE)
        }
    }

    companion object {
        private var INSTANCE: SharedPreferencesManger? = null

        fun getINSTANCE(context: Context?): SharedPreferencesManger? {
            if (INSTANCE == null) {
                INSTANCE = SharedPreferencesManger(context!!)
            }
            return INSTANCE
        }
    }


    fun loadUserData(): ClientData? {
        val loginData: ClientData?
        val gson = Gson()
        loginData = gson.fromJson(restoreStringValue(USER_DATA), ClientData::class.java)
        return loginData
    }

    fun clean() {
        getINSTANCE(context)
        if (sharedPreferences != null) {
            val editor = sharedPreferences!!.edit()
            editor.clear()
            editor.apply()
        }
    }

    fun setPreference(key: String, value: String) {
        val editor: SharedPreferences.Editor = sharedPreferences!!.edit()
        editor.putString(key, value)
        editor.apply()
    }


    fun getApiToken(): String {
        return sharedPreferences!!.getString(API_TOKEN, "")!!
    }

    fun setPreference(key: String, value: Int) {
        val editor: SharedPreferences.Editor = sharedPreferences!!.edit()
        editor.putInt(key, value)
        editor.apply()
    }

    fun setPreference(key: String, value: Boolean) {
        val editor: SharedPreferences.Editor = sharedPreferences!!.edit()
        editor.putBoolean(key, value)
        editor.apply()
    }

    fun setPreference(key: String, value: Any) {
        val editor: SharedPreferences.Editor = sharedPreferences!!.edit()
        val gson = Gson()
        val stringData: String = gson.toJson(value)
        editor.putString(key, stringData)
        editor.apply()
    }

    open fun restoreStringValue(key: String?): String? {
        return sharedPreferences!!.getString(key, "Empty")
    }

    fun restoreIntValue(key: String?): Int? {
        return sharedPreferences!!.getInt(key, 0)
    }

    fun restoreBooleanValue(key: String?): Boolean? {
        return sharedPreferences!!.getBoolean(key, false)
    }
}